tests/Microsoft.Identity.Test.Unit/CoreTests/RegionDiscoveryProviderTests.cs

```diff
@@ -113,31 +117,65 @@ public async Task ErrorResponseFromLocalImdsAsync()
 
                 Assert.Fail("Exception should be thrown.");
             }
-            catch (MsalClientException e)
+            catch (MsalServiceException e)
             {
                 Assert.IsNotNull(e);
                 Assert.AreEqual(MsalError.RegionDiscoveryFailed, e.ErrorCode);
                 Assert.AreEqual(MsalErrorMessage.RegionDiscoveryFailed, e.Message);
             }
         }
 
-        private void AddMockedResponse(HttpResponseMessage responseMessage)
+        [TestMethod]
+        public async Task UpdateApiversionWhenCurrentVersionExpiresForImdsAsync()
```

Comment:
Are thjere any unit tests that cover the case where a MsalError.RegionDiscoveryFailed exception is thrown?

Reply:
Added more unit tests

Reply:
Added more unit tests covering failure scenarios